features/net/network-socket/nsapi_dns.cpp

```diff
@@ -19,21 +19,29 @@
 #include <string.h>
 #include <stdlib.h>
 
+#define CLASS_IN 1
+
+#define RR_A 1
+#define RR_AAAA 28
 
 // DNS options
-#define DNS_BUFFER_SIZE 256 
+#define DNS_BUFFER_SIZE 512
```

Comment:
The DNS spec says 512 bytes. We reduced it from 1024 to 512 during devlopment, but when we saw that someone else had been touching the DNS code (aargh), we saw you'd reduced it further to 256. Left it at 512 during conflict resolution.

Not seen any failures, but I'm not totally confident about using IPv6 DNS with half-sized buffers.

(We also made the original DNS reply parsing resilient to truncated results, but had to abandon it - wasn't straightforward to refit to the new code, which doesn't consider packet length at all).